src/OrchardCore/OrchardCore.Recipes.Core/Services/RecipeExecutor.cs

```diff
@@ -197,21 +198,29 @@ private void EvaluateJsonTree(IScriptingManager scriptingManager, RecipeExecutio
                     {
                         EvaluateJsonTree(scriptingManager, context, array[i]);
                     }
+
                     break;
                 case JTokenType.Object:
                     foreach (var property in (JObject)node)
                     {
                         EvaluateJsonTree(scriptingManager, context, property.Value);
                     }
-                    break;
 
+                    break;
                 case JTokenType.String:
-
+                    const char scriptSeparator = ':';
                     var value = node.Value<string>();
+                    var scriptSeparatorIndex = value.IndexOf(scriptSeparator);
```

Comment:
This should be inside the while loop

Reply:
It was there, seems I copied accidently, thanks